[Alerting] Add UI indicators for rules with less than configured minimum schedule interval

ymao1 opened the pull request:
Resolves https://github.com/elastic/kibana/issues/126409

## Summary

* Adds info indicator on rules list for rules that are configured to run below the `minimumScheduleInterval`
* If user is able to edit this rule, clicking on this indicator will open the `Edit rule` flyout
* Adds info toast to rule details page on load for rules that are configured to run below the `minimumScheduleInterval`
* If user is able to edit this rule, toast will include action button that, when clicked, opens the `Edit rule` flyout

https://user-images.githubusercontent.com/13104637/159571932-4724e2c7-1fb9-43c1-afc4-2a63c78a9864.mp4



### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

mdefazio commented on x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx:
```diff
@@ -609,7 +619,59 @@ export const RulesList: React.FunctionComponent = () => {
+                            onRuleEdit(item);
+                          }
+                        }}
+                        iconType="iInCircle"
```
Let's change this to `flag`.  After seeing this more in action the `iInCircle` doesn't feel correct.

mdefazio reviewed (changes requested):
Thank you for the request! A few small edits!

ymao1 commented:
@mdefazio Addressed your comments in this commit: https://github.com/elastic/kibana/pull/128254/commits/efeabed7b999716b6d1e5604889c0805a26b2748

Updated screenshot:
<img width="1305" alt="Screen Shot 2022-03-23 at 2 53 42 PM" src="https://user-images.githubusercontent.com/13104637/159774894-3240db94-d1aa-485e-ac18-0326ac264d13.png">

mdefazio reviewed (approved):
thanks! LGTM

ymao1 commented:
@elasticmachine merge upstream
